modules/index/src/main/java/org/apache/ignite/internal/index/IndexManager.java

```diff
@@ -0,0 +1,263 @@
+import org.apache.ignite.lang.ErrorGroups.Table;
+import org.apache.ignite.lang.IgniteInternalException;
+import org.apache.ignite.lang.IndexAlreadyExistsException;
+import org.apache.ignite.lang.NodeStoppingException;
+import org.apache.ignite.lang.TableNotFoundException;
+
+/**
+ * An Ignite component that is responsible for handling index-related commands like CREATE or DROP
+ * as well as managing indexes lifecycle.
+ */
+public class IndexManager implements IgniteComponent {
+    private static final IgniteLogger LOG = Loggers.forClass(IndexManager.class);
+
+    private final TableManager tableManager;
+
+    private final Map<UUID, Index> indices = new ConcurrentHashMap<>();
+
+    /** Busy lock to stop synchronously. */
+    private final IgniteSpinBusyLock busyLock = new IgniteSpinBusyLock();
+
+    /** Prevents double stopping of the component. */
+    private final AtomicBoolean stopGuard = new AtomicBoolean();
+
+    /**
+     * Constructor.
+     *
+     * @param tableManager Table manager.
+     */
+    public IndexManager(
+            TableManager tableManager
+    ) {
+        this.tableManager = Objects.requireNonNull(tableManager, "tableManager");
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    public void start() {
+        LOG.info("Index manager started");
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    public void stop() throws Exception {
+        LOG.debug("Index manager is about to stop");
+
+        if (!stopGuard.compareAndSet(false, true)) {
+            LOG.debug("Index manager already stopped");
+
+            return;
+        }
+
+        busyLock.block();
+
+        LOG.info("Index manager stopped");
+    }
+
+    /**
+     * Creates index from provided configuration changer.
+     *
+     * @param schemaName A name of the schema to create index in.
+     * @param indexName A name of the index to create.
+     * @param tableName A name of the table to create index for.
+     * @param indexChange A consumer that suppose to change the configuration in order to provide description of an index.
+     * @return A future represented the result of creation.
+     */
+    // TODO: https://issues.apache.org/jira/browse/IGNITE-17474
+    // Validation of the index name uniqueness is not implemented, because with given
+    // configuration hierarchy this is a bit tricky exercise. Given that this hierarchy
+    // is subject to change in the future, seems to be more rational just to omit this
+    // part for now
+    public CompletableFuture<Index> createIndexAsync(
+            String schemaName,
+            String indexName,
+            String tableName,
+            Consumer<TableIndexChange> indexChange
+    ) {
+        if (!busyLock.enterBusy()) {
+            return CompletableFuture.failedFuture(new NodeStoppingException());
+        }
+
+        LOG.debug("Going to create index [schemaName={}, tableName={}, indexName={}]", schemaName, tableName, indexName);
```

Comment:
maybe short version of parameters name will be better here and below? Like a schema={}, table={}, index={}

Reply:
ok, fixed